### PR TITLE
fix: Enhanced multiline support in TextMate grammar for source files

### PR DESCRIPTION
**Pull request by jirimosinger**
Set the native VS Code option "Semantic Highlighting" to "False" to suppress semantics tokens and to view the TextMate grammar highlighting:

image

### CODE REVIEW

File: clients/vscode-hlasmplugin/syntaxes/hlasm.tmLanguage.json [hunk 1]
```diff
@@ -233,92 +170,174 @@
...
+    "sequenceSymbolOperandField": {
+      "match": "\\.[a-zA-Z@$#_0-9]+(?=\\s|$)(?<=^.{0,71})",
+      "name": "keyword.control.hlasm"
+    },
+    "label": {
+      "match": "^\\S+(?<=^.{0,71})",
       "name": "support.class.hlasm"
     },
+    "instruction": {
+      "match": "((?<=^\\S*)\\s+|(?<=^\\S*\\s+))\\S+(?<=^.{0,71})",
```
Comment (slavek-kucera) [inline clients/vscode-hlasmplugin/syntaxes/hlasm.tmLanguage.json:315]: I don't really understand the beginning of the pattern. It tries to match the space, but also just check whether it is there??

File: clients/vscode-hlasmplugin/syntaxes/hlasm.tmLanguage.json [hunk 2]
```diff
@@ -233,92 +170,174 @@
...
+      ]
+    },
+    "sequenceSymbolNameField": {
+      "match": "^\\.\\S+(?<=^.{0,71})",
+      "name": "keyword.control.hlasm"
+    },
+    "sequenceSymbolOperandField": {
+      "match": "\\.[a-zA-Z@$#_0-9]+(?=\\s|$)(?<=^.{0,71})",
+      "name": "keyword.control.hlasm"
+    },
```
Comment (slavek-kucera) [inline clients/vscode-hlasmplugin/syntaxes/hlasm.tmLanguage.json:309]: should we go for the strict sequence symbol rules? `\\.[A-Za-z@#$_][A-Za-z@#$_0-9]*`

Comment (jirimosinger) [reply clients/vscode-hlasmplugin/syntaxes/hlasm.tmLanguage.json:309]: Sure!

File: clients/vscode-hlasmplugin/syntaxes/hlasm.tmLanguage.json (hunk 1 above)
Comment (jirimosinger) [reply clients/vscode-hlasmplugin/syntaxes/hlasm.tmLanguage.json:315]: There is a case where the remaining text is either ` INSTR OP ...` or `INSTR OP ...` and this patter tries to match both versions.

Nevertheless, I'll re-verify if it's really necessary to have both patterns there

File: clients/vscode-hlasmplugin/syntaxes/hlasm.tmLanguage.json [hunk 3]
```diff
@@ -19,212 +20,148 @@
...
-          "include": "#stringType"
+          "include": "#quoteType"
         },
         {
-          "include": "#quoteType"
+          "include": "#openEndedArgs"
         },
         {
-          "match": "(\\S*)(.*)",
+          "match": "([^\\s=(),'*/+-]+)(?<=^.{0,71})",
```
Comment (slavek-kucera) [inline clients/vscode-hlasmplugin/syntaxes/hlasm.tmLanguage.json:164]: What is the significance of `s` here? `\\s` in the `[^]` pattern probably does not do what was intended

Comment (jirimosinger) [reply clients/vscode-hlasmplugin/syntaxes/hlasm.tmLanguage.json:164]: In this case it should force the TextMate grammar to iterate over arguments from `operandsAndRemarks` and from `parenthesis` one by one without consuming any separating spaces, commas, etc.

**Comment by slavek-kucera** [comment]
- It looks like sequence numbers take on different colors if the instruction has no operands
- if there is text in the columns 81+ the whole line becomes red

**Comment by jirimosinger** [comment]
Seems like a rule is missing for the first point.

What would you suggest for the second one? Should only the text starting at column 81 become red?

**Comment by slavek-kucera** [comment]
Yes, only 81+ being red seems like a better approach. Or 73+ to be consistent with what we then generate from the server.

**Comment by slavek-kucera** [comment]
Would it be a lot of trouble to also detect basic asm variables `&[A-Za-z@#$_][A-Za-z@#$_0-9]*`?

**Comment by slavek-kucera** [comment]
Maybe one last detail - only the last sequence symbol in the following statement is highlighted
```
         AGO   (&A).HERE,.THERE,.EVERYWHERE
```

**Review by slavek-kucera — APPROVED** [review]

**Review by slavek-kucera — APPROVED** [review]

**Pull request merged**